bin/spiced/src/main.rs

```diff
@@ -49,9 +60,85 @@ fn main() {
         return;
     }
 
-    if let Err(err) = tokio_runtime.block_on(start_runtime(args)) {
+    // Create a broadcast channel to signal shutdown across all threads
+    let (shutdown_tx, _) = broadcast::channel::<()>(32); // Increase capacity to ensure all components can receive signals
```

Comment:
I'm under impression is that capacity is number of messages here, not max receivers